Adding Simulation Experiment - Importance of Compression Feature Number (#188)
greenelab/BioBombe: 11.simulation-feature-number/README.md

# Simulate Compressed Feature Extraction

## Simulated Data

In the following analysis module, we generate data with two groups of artificial signals.
We generate a 10,000 x 10 matrix where two groups of three features each are highly correlated.
The remaining features are sampled with random Gaussian noise.

The two groups of features have different correlation structures.
The first group (features 1, 2, and 3) is simulated to be highly correlated with a range 0.90 - 0.95 (Pearson's correlation).
The second group of features (features 5, 6, 7) is simulated to be correlated to a slightly lesser degree (range: 0.85 - 0.90).

| Group | Correlated Features | Correlation Range |
| :---- | :------------------ | :---------------- |
| 1 | 1, 2, 3 | 0.90 - 0.95 |
| 2 | 5, 6, 7 | 0.85 - 0.90 |

## Experimental Design

> The experiment is to apply the BioBombe suite of algorithms (PCA, ICA, NMF, DAE, VAE) across 6 different latent dimensionalities (k = 1, 2, 3, 4, 5, 6).

Reviewer: ```suggestion
For this experiment we apply the BioBombe suite of algorithms (PCA, ICA, NMF, DAE, VAE) across 6 different latent dimensionalities (k = 1, 2, 3, 4, 5, 6).
```

> 

Reviewer: ```suggestion
```

We fit each model and extract the resulting weight matrix.
We observe the contributions (weights or importance scores) of each input raw feature to each compressed feature.

> Our goal is to determine the **top compressed feature number** that represents each simulated feature in both groups.

Reviewer: ```suggestion
Our goal is to determine the **number** of the compressed feature that best represents each simulated feature in both groups.
```
